### PR TITLE
Bit12

### PR DESCRIPTION
**Pull request by jennyfolkesson**
Adding support for:

uint12 images
pixel size changes in get_spot_intensity

### CODE REVIEW

**Comment by bryantChhun** [comment]
I'm running the repo on the latest nautilus data and finding it takes approx 14 seconds per image.  Is this what you've observed @jennyfolkesson ?

File: array_analyzer/transform/array_generation.py [hunk 1]
```diff
@@ -88,6 +88,8 @@ def get_spot_intensity(coords, im, background, params, search_range=2):
     # make spot size always odd
     spot_size = 2 * int(0.3 * spot_width / pix_size) + 1
     bbox_width = bbox_height = spot_size
+    # Strel disk size for spot segmentation
+    disk_size = int(np.rint(spot_size / 2.5))
```
Comment (smguo) [inline array_analyzer/transform/array_generation.py:92]: Yes the filter size needs to change with the spot size... thanks for fixing this

File: array_analyzer/transform/point_registration.py [hunk 2]
```diff
@@ -246,4 +246,5 @@ def particle_filter(fiducial_coords,
 
     # Generate transformation matrix
     t_matrix = get_translation_matrix(particle)
-    return t_matrix, min_dist
+    reg_dist = min_dist / (fiducial_coords.shape[0] - nbr_outliers)
+    return t_matrix, reg_dist
```
Comment (smguo) [inline array_analyzer/transform/point_registration.py]: Maybe call it something like `regis_error` or `regis_dist`? Took me a while to figure out what the name `reg_dist` means...  updating the docstring would help too :)

File: array_analyzer/utils/io_utils.py [hunk 3]
```diff
@@ -39,6 +39,28 @@ def read_gray_im(im_path):
...
+        "Must use integer images, not {}".format(im.dtype)
+    max_intensity = np.iinfo(im.dtype).max
+    # Check if a 16 bit image is actually 12 bit
+    if max_intensity == 65535 and im.max() < 4096:
+        max_intensity = 4095
+    assert max_intensity in {255, 4095, 65535}, \
+        "Image must be uint 8, 12, or 16, not have max {}".format(max_intensity)
+    return max_intensity
+
+
```
Comment (smguo) [inline array_analyzer/utils/io_utils.py:63]: 👍

File: tests/utils/io_utils_test.py [hunk 4]
```diff
@@ -15,6 +16,32 @@ def test_no_im(image_dir):
...
+    max_intensity = io_utils.get_max_intensity(im)
+    assert max_intensity == 65535
+
+
+def test_get_max_intensity_float():
+    im = np.zeros((2, 3), dtype=np.float32)
+    with pytest.raises(AssertionError):
+        io_utils.get_max_intensity(im)
+
+
```
Comment (smguo) [inline tests/utils/io_utils_test.py:44]: 💯

**Review by smguo — APPROVED** [review]
I went through the code changes and they look great @jennyfolkesson! I only have one minor comment on variable name and docstring.

**Pull request merged**